### PR TITLE
[kube-prometheus-stack] Configure possibility of providing additional rules to prometheus cluster role

### PR DESCRIPTION
**Pull request by dieterdemeyer**
#### What this PR does / why we need it:

This change will fix the Unhealthy Target Group `kubernetes-nodes` which will scrape node metrics from the `/proxy` endpoint.

#### Checklist
- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — APPROVED** [review]

**Review by bismarck — CHANGES_REQUESTED** [review]
@dieterdemeyer There's no `kubernetes-nodes` job in the kube-prometheus-stack chart. I see one in the [prometheus chart](https://github.com/prometheus-community/helm-charts/blob/4f4574545ccd5549bd67ef495b8540ad40e3310c/charts/prometheus/values.yaml#L1281-L1315)?

**Comment by dieterdemeyer** [comment]
@bismarck ,

Yes, you are correct.

We actually use custom `values.yaml` file which specifies a number of additional jobs and `kubernetes-nodes` is one of them.
We currently use the `prometheus-operator` helm chart and the additional proxy endpoint is listed in the cluster role defined there but is missing in the `kube-prometheus-stack` helm chart.

**Comment by vsliouniaev** [comment]
I'm against changing the default RBAC scope to support a custom configuration. The chart allows a different role to be used that is not provisioned by the chart, so that is probably your best option without any modifications.

If you do want to extend functionality here to support additions to the RBAC scope I think that would be a reasonable extension.

**Comment by dieterdemeyer** [comment]
@vsliouniaev , I've updated the PR. Is this more in line of what you meant as an extension ?

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 12.2.3
+version: 12.2.4
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:21]: this is a new feature now and should raise minor version.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml (hunk 1 above)
Comment (dieterdemeyer) [reply charts/kube-prometheus-stack/Chart.yaml]: Version has been bumped to 12.3.1.

**Review by monotek — CHANGES_REQUESTED** [review]
version should be 12.5.0

**Comment by dieterdemeyer** [comment]
@monotek , version has been bumped.

**Review by monotek — APPROVED** [review]

File: charts/kube-prometheus-stack/templates/prometheus/clusterrole.yaml [hunk 2]
```diff
@@ -24,4 +24,7 @@ rules:
   verbs: ["get", "list", "watch"]
 - nonResourceURLs: ["/metrics", "/metrics/cadvisor"]
   verbs: ["get"]
+{{- if and .Values.prometheus.enabled .Values.prometheus.additionalRulesForClusterRole }}
```
Comment (bismarck) [inline charts/kube-prometheus-stack/templates/prometheus/clusterrole.yaml:27]: `.Values.prometheus.enabled` is checked at the top of the file, so it isn't needed here.

**Review by bismarck — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/templates/prometheus/clusterrole.yaml (hunk 2 above)
Comment (dieterdemeyer) [reply charts/kube-prometheus-stack/templates/prometheus/clusterrole.yaml]: Unnecessary check has been removed.

**Review by bismarck — APPROVED** [review]
LGTM

**Pull request merged**